sqld/src/replication/primary/logger.rs

```diff
@@ -897,6 +897,14 @@ fn checkpoint_db(data_path: &Path) -> anyhow::Result<()> {
             );
             Ok(())
         })?;
+        // turn off auto_checkpointing - we'll use a fiber to checkpoint in time steps instead
+        let rc = rusqlite::ffi::sqlite3_wal_autocheckpoint(conn.handle(), 0);
+        if rc != 0 {
+            return Err(anyhow!(
```

Comment:
nitpick/hint: this is equivalent to shorter: `bail!("Failed to disable WAL autocheckpoint - error code: {rc}")`. Also applies to other conditions below

Reply:
yeah, and clippy should complain

Reply:
clippy doesn't complain. I've changed it anyway.